src/util/util_errors.h

```diff
@@ -211,6 +211,10 @@ enum sssd_errors {
 #define ELIBBAD EBADF
 #endif
 
+#ifndef ENODATA
+#define ENODATA ESHUTDOWN
```

Comment:
Again, `ENODATA` has an internal semantic, and re-using another error code can, in theory, trigger unexpected behavior.

Maybe `ENOMSG` would fit better.

Reply:
Also I suggest to keep consistent style and to either use `&& defined(...)` everywhere on nowhere.